compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block_variables.rs

```diff
@@ -13,21 +13,21 @@ use crate::{
     ssa::ir::{
         dfg::DataFlowGraph,
         types::{CompositeType, Type},
-        value::ValueId,
+        value::{RawValueId, ResolvedValueId, ValueId},
     },
 };
 
 use super::brillig_fn::FunctionContext;
 
 #[derive(Debug, Default)]
 pub(crate) struct BlockVariables {
-    available_variables: HashSet<ValueId>,
+    available_variables: HashSet<RawValueId>,
```

Comment:
It seems this could be `ResolvedValueId`. I just tried it with a small refactor. I noticed we only insert resolved values in this set, so it could get us a bit more type-safety.

Reply:
Argh, I thought I sent a lengthy reply to this, but it's lost :-/ 
Good spot, thanks for looking at it with a fine comb! You're right, we can use `ResolvedValueId` here, I pushed a commit to do so. 

The reason I didn't was that in general I wasn't sure whether using `ResolvedValueId` in data structures is safe even when we always have it available at the insertion side: what if later it becomes stale? Say we have `v0`, replaced by `v10` and inserted into a map, then replaced again by `v20`; looking up by `v20`, or iterating over the keys and comparing them against `v20` wouldn't find the entry, even though the compiler wouldn't complain.

But this module is `brillig_gen`, so I suppose there won't be any more replacements. I tried to nudge it towards this way by changing the API methods to take `ResolveValueId`, but wanted to be conservative inside. It seems like it was easy to go the last step, there was only 1 method left.